### PR TITLE
SCons: Fix build with `p` alias or platform auto-detection

### LINKED ISSUE #44448: Windows build fails: cannot open include file 'netinet/in.h'
**Godot version:**
master b07a3f503b75831414f92d2cc6886d045eeb96fb

**OS/device including version:**
Windows 10

**Issue description:**
Windows build fails, even with clean build. Worked before #44433 was merged. cc @akien-mga 

```
C:\Users\mbr\Documents\Projects\godot\thirdparty\wslay\wslay_net.h(38): fatal error C1083: Cannot open include file: 'netinet/in.h': No such file or directory
C:\Users\mbr\Documents\Projects\godot\thirdparty\wslay\wslay_net.h(38): fatal error C1083: Cannot open include file: 'netinet/in.h': No such file or directory
scons: *** [thirdparty\wslay\wslay_net.windows.opt.tools.64.obj] Error 2
scons: *** [thirdparty\wslay\wslay_event.windows.opt.tools.64.obj] Error 2
scons: building terminated because of errors.
```

### PR DESCRIPTION
**Pull request by akien-mga**
TL;DR: What a mess... :100: would do it again the same way :face_with_head_bandage: 

---

Fixes a pre-existing bug that #44433 exposed.

It's pretty hacky, but we use `platform` in `env` both as an optional command
line option (instead it can be autodetected, or passed via the `p` alias, and
on Linux it might be overridden if you pass one of the convenience alias
values), and as the reference value for what platform we're building on.

Thus we override `env_base["platform"]` with the autodetected or validated
platform, but any call to `opts.Update(env_base)` overrides it with the
original command line option... causing e.g. #44448.

The proper fix would be to refactor all this so that we don't reuse
`env["platform"]` for platform detection (it could instead be e.g.
`env.platform` as a member variable which holds the validated value),
but for now I'm tapering over the immediate breakage.

Fixes #44448 and other breakages induced by #44433.

### CODE REVIEW

**Comment by akien-mga** [comment]
> The proper fix would be to refactor all this so that we don't reuse
> `env["platform"]` for platform detection (it could instead be e.g.
> `env.platform` as a member variable which holds the validated value),
> but for now I'm tapering over the immediate breakage.

Note that while this is easy to implement and would be much cleaner, it would also break compatibility for custom modules or platforms that rely on `env["platform"]` too.

Another option is to renaming the `platform` command line option and keep `env["platform"]` as a non-option validated platform name.

**Comment by bruvzg** [comment]
macOS build is fixed.

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.
